cdk/eks/lib/stacks/iam-stack.ts

```diff
@@ -55,6 +55,7 @@ export class IAMStack extends Stack {
         ManagedPolicy.fromAwsManagedPolicyName('AmazonEC2ContainerRegistryReadOnly'),
         ManagedPolicy.fromAwsManagedPolicyName('AmazonEKS_CNI_Policy'),
         ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSWorkerNodePolicy'),
+        ManagedPolicy.fromAwsManagedPolicyName('AWSXrayFullAccess'),
```

Comment:
Can you use the service account to add the permission to the relevant pod rather than the whole cluster?